app/components/draft_post_header/draft_post_header.tsx

```diff
@@ -0,0 +1,162 @@
+import {getUserTimezone} from '@utils/user';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type PostModel from '@typings/database/models/servers/post';
+import type UserModel from '@typings/database/models/servers/user';
+
+type Props = {
+    channel: ChannelModel;
+    draftReceiverUser?: UserModel;
+    updateAt: number;
+    rootId?: PostModel['rootId'];
+    testID?: string;
+    currentUser?: UserModel;
+    isMilitaryTime: boolean;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            display: 'flex',
+            flexDirection: 'row',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+        },
+        infoContainer: {
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        channelInfo: {
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        category: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75, 'SemiBold'),
+            marginRight: 8,
+        },
+        categoryIconContainer: {
+            width: 24,
+            height: 24,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+            padding: 4,
+            borderRadius: 555,
+        },
+        profileComponentContainer: {
+            marginRight: 6,
+        },
+        displayName: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75, 'SemiBold'),
+        },
+        time: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+        },
+    };
+});
+
+const DraftPostHeader: React.FC<Props> = ({
+    channel,
+    draftReceiverUser,
+    updateAt,
+    rootId,
+    testID,
+    currentUser,
+    isMilitaryTime,
+}) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const isChannelTypeDM = channel.type === General.DM_CHANNEL;
+
+    let headerComponent: ReactNode = null;
+    const profileComponent = draftReceiverUser ? <ProfileAvatar author={draftReceiverUser}/> : (
+        <View style={style.categoryIconContainer}>
+            <CompassIcon
+                color={changeOpacity(theme.centerChannelColor, 0.64)}
+                name='globe'
+                size={16}
+            />
+        </View>);
+
+    if (rootId) {
+        headerComponent = (
+            <View style={style.channelInfo}>
+                <FormattedText
+                    id='channel_info.thread_in'
+                    defaultMessage={'Thread in:'}
+                    style={style.category}
+                />
+                <View style={style.profileComponentContainer}>
+                    {profileComponent}
+                </View>
+            </View>
```

Comment:
Can we create an inline component instead?  It seems between the three if-else conditions, the things that are different is the `id` and `defaultMessage.`

Reply:
Done.